conformance/utils/roundtripper/roundtripper.go

```diff
@@ -18,21 +18,23 @@ package roundtripper
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"net/http/httputil"
 	"net/url"
 	"regexp"
-
 	"sigs.k8s.io/gateway-api/conformance/utils/config"
 )
 
 // RoundTripper is an interface used to make requests within conformance tests.
 // This can be overridden with custom implementations whenever necessary.
 type RoundTripper interface {
 	CaptureRoundTrip(Request) (*CapturedRequest, *CapturedResponse, error)
+	CaptureTLSRoundTrip(Request, []byte, []byte, string) (*CapturedRequest, *CapturedResponse, error)
```

Comment:
Any chance we can reuse CaptureRoundTrip just with a few more options in the Request? At first glance it looks like CaptureTLSRoundTrip includes a lot of the same code.

Reply:
I did it this way because I didn't want to change the Request, but I'll look into it.

Reply:
Done.